### DISCUSSION TITLE
Error handling seems a little cumbersome (improve DX?)

### DISCUSSION THREAD

**Discussion by harrywebdev** [evoluhq/evolu#60] [Ideas]
Consider this example:

```
import { getError, subscribeError } from "evolu"

// show DB errors
useEffect(() => {
  const notifyOnError = () => {
    const error = getError()
    if (error === null) {
      return
    }

    // use message, if it exists
    if (error.error.type === "UnknownError") {
      // wrap on the error to get to the actual error
      setNotificationMessage(`Error: ${error.error.error.message}`)
    } else {
      setNotificationMessage("Oops, something went wrong.")
    }
  }

  return subscribeError(notifyOnError)
}, [])
```

Let's look at the types:

```
export const getError: IO<EvoluError | null> = () => lastErrorRef.read();

export interface EvoluError {
  readonly type: "EvoluError";
  readonly error:
    | TimestampDuplicateNodeError
    | TimestampDriftError
    | TimestampCounterOverflowError
    | TimestampParseError
    | StringMaxLengthError
    | UnknownError
    | SyncError
    | SQLiteError;
}
```

so the `EvoluError` is a wrapper and its property `error` contains the actual error. Now, this error contains various types, ranging from errors without an actual message (their type is the basis for the message) like:

```
export interface TimestampDuplicateNodeError {
  readonly type: "TimestampDuplicateNodeError";
  readonly node: NodeId;
}
```

to an unknown error with a message (wrapped under the `error` property)
```
export interface UnknownError {
  readonly type: "UnknownError";
  readonly error: TransferableError;
}

interface TransferableError {
  readonly message: string;
  readonly stack: string | undefined;
}
```

Maybe for production cases, the `EvoluError.error.type` is enough to produce a feedback message to the user, and the wrap to get a message from `UnknownError` is good enough for debugging purposes. Maybe, we could do better in how we surface errors - the way I came across this was when I struggled with a correct SQL query and I was getting at the actual DB error message.

**Comment by steida** [comment]
If you see how it can be improved, describe a solution you would like to see based on your experience.

**Comment by steida** [comment]
Already fixed
